pinot-common/src/main/java/org/apache/pinot/common/utils/SegmentUtils.java

```diff
@@ -39,14 +39,30 @@ private SegmentUtils() {
   @Nullable
   public static Integer getRealtimeSegmentPartitionId(String segmentName, String realtimeTableName,
       HelixManager helixManager, @Nullable String partitionColumn) {
+    return getRealtimeSegmentPartitionId(
+        segmentName, realtimeTableName, helixManager, partitionColumn, null);
+  }
+
+  @Nullable
+  public static Integer getRealtimeSegmentPartitionId(String segmentName, String realtimeTableName,
+      SegmentZKMetadata segmentZKMetadata) {
+    return getRealtimeSegmentPartitionId(segmentName, realtimeTableName, null, null, segmentZKMetadata);
```

Comment:
This method should contain code from line 52-67 in the original code. Change the original method to keep line 42-51 and then call this new method